core/src/main/java/org/testcontainers/utility/ContainerReaper.java

```diff
@@ -0,0 +1,80 @@
+package org.testcontainers.utility;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.DockerException;
+import com.github.dockerjava.api.InternalServerErrorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.DockerClientFactory;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+/**
+ * Component that responsible for container removal and automatic cleanup of dead containers at JVM shutdown.
+ */
+public final class ContainerReaper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ContainerReaper.class);
```

Comment:
Good point - let's make that part of a project wide refactor, though, as other classes all use explicit declaration right now.